modules/services/monitoring/prometheus-node-exporter.nix

```diff
@@ -0,0 +1,81 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    concatStringsSep
+    escapeShellArgs
+    getExe
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    types
+  ;
+
+  cfg = config.services.prometheus-node-exporter;
+in {
+  options = {
+    services.prometheus-node-exporter = {
+      enable = mkEnableOption "Prometheus Node exporter";
+
+      package = mkPackageOption pkgs "prometheus-node-exporter" { };
+
+      listenAddress = mkOption {
```

Comment:
I’m not sure why NixOS has both `port` and `listenAddress`. If one is (soft‐)deprecated then we can probably just keep the best practice. Otherwise maybe we should support both.

Reply:
Generalization of the interface, but since node-exporter is written in Go it can work with specifications like `:9100`, which a normal BSD style sockaddr struct cannot.